src/main/java/org/springframework/analytics/rest/controller/AggregateCounterController.java

```diff
@@ -71,31 +63,45 @@ public AggregateCounterController(AggregateCounterRepository repository) {
 
 	/**
 	 * Retrieve information about a specific aggregate counter.
+	 *
+	 * @param name  counter name
+	 * @return {@link AggregateCounterResource}
+	 *
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.GET)
 	public AggregateCounterResource display(@PathVariable("name") String name) {
 		AggregateCounter counter = repository.findOne(name);
 		if (counter == null) {
-			throw new MetricsMvcEndpoint.NoSuchMetricException(name);
+			throw new IllegalArgumentException(name);
```

Comment:
I think we need to find some other exception to throw, the reason is that in the REST layer that calls these methods (hence the use of `AggregateCounterResource`) when the counter is not found, needs to return `@ResponseStatus(HttpStatus.NOT_FOUND)` .    Since that is handled in general `@ControllerAdvice` we need a more specific exception to indicate the counter was not found.  ATM there isn't a general `ResourceNotFound` exception we can use.  I would suggest adding back `NoSuchMetricException`.